src/aiida/engine/daemon/execmanager.py

```diff
@@ -435,56 +436,108 @@ async def stash_calculation(calculation: CalcJobNode, transport: Transport) -> N
+                else:
+                    EXEC_LOGGER.debug(f'stashed {source_filepath} to {target_filepath}')
+
+        remote_stash = RemoteStashFolderData(
+            computer=calculation.computer,
+            target_basepath=str(target_basepath),
+            stash_mode=StashMode(stash_mode),
+            source_list=source_list,
+        ).store()
+
+    elif stash_mode in [
+        StashMode.COMPRESS_TAR.value,
+        StashMode.COMPRESS_TARBZ2.value,
+        StashMode.COMPRESS_TARGZ.value,
+        StashMode.COMPRESS_TARXZ.value,
+    ]:
+        # stash_mode values are identical with compression_format in transport plugin:
+        # 'tar', 'tar.gz', 'tar.bz2', or 'tar.xz'
+        compression_format = stash_mode
+        file_name = uuid
+        dereference = stash_options.get('dereference', False)
+        target_basepath = Path(stash_options['target_base'])
+        authinfo = calculation.get_authinfo()
+        aiida_remote_base = authinfo.get_workdir().format(username=transport.whoami())
+
+        target_destination = str(target_basepath / file_name) + '.' + compression_format
+
+        remote_stash = RemoteStashCompressedData(
+            computer=calculation.computer,
+            target_basepath=target_destination,
+            stash_mode=StashMode(stash_mode),
+            source_list=source_list,
+            dereference=dereference,
+        )
 
-    uuid = calculation.uuid
-    source_basepath = Path(calculation.get_remote_workdir())
-    target_basepath = Path(stash_options['target_base']) / uuid[:2] / uuid[2:4] / uuid[4:]
-
-    for source_filename in source_list:
-        if transport.has_magic(source_filename):
-            copy_instructions = []
-            for globbed_filename in await transport.glob_async(source_basepath / source_filename):
-                target_filepath = target_basepath / Path(globbed_filename).relative_to(source_basepath)
-                copy_instructions.append((globbed_filename, target_filepath))
+        source_list_abs = [source_basepath / source for source in source_list]
+
+        try:
+            await transport.compress_async(
+                format=compression_format,
+                remotesources=source_list_abs,
+                remotedestination=target_destination,
+                root_dir=aiida_remote_base,
+                overwrite=False,
+                dereference=dereference,
+            )
+        except (OSError, ValueError) as exception:
+            EXEC_LOGGER.warning(f'failed to stash {source_list} to {target_destination}: {exception}')
+            return
+            # note: if you raise here, you triger the exponential backoff
+            # and if you don't raise, it appears as successful in verdi process list: Finished [0]
+            # open an issue to fix this
+            # raise exceptions.RemoteOperationError(f'failed '
+            # 'to compress {source_list} to {target_destination}: {exception}')
```

Comment:
I guess we want here to start the exponential backoff mechanism only if it fails due to some connection problem, right? Maybe we check the error if it is due to connection issues (somehow the transport plugin needs to provide the kind of error that is due to connection issues). If it is not due to a connection issues we just fail here, then we can put the process state as Excepted. You are right here we cannot do it we have to change `task_stash_job` in  `engine/processes/calcjobs/tasks.py`. Can you open an issues then we can move the discussion to there.